### PR TITLE
Add support for embedded mode

### PR DESCRIPTION
**Pull request by lemieux**
I did a couple of things in this PR : 

- created a new test page for embedded mode (available at `/embedded`)
- ported over our update eslint and esformatter rules
- adjusted CSS to split between normal and embedded mode
- removed jshint stuff
- decoupled `init` from rendering
- used proper unmount mecanism from React to clean up on destroy

Tests are currently missing, but feel free to start reviewing and I'll add them.

The CSS split is kinda hackish, but I think it's good enough for a first version. I'll spend time on it this week to clean it up.

@spasiu @alavers @dannytranlx @Mario54

### CODE REVIEW

**Comment by lemieux** [comment]
We might want display a loading spinner of some sort when `render` happens before `init` (which could happen easily and that's okay really).

**Comment by lemieux** [comment]
It also needs documentation.

**Comment by spasiu** [comment]
Damnit man, that's awesome. I'm going to point them at this branch for now. And read more.

File: src/js/components/header.jsx [hunk 1]
```diff
@@ -46,19 +46,22 @@ export class HeaderComponent extends Component {
...
-        const closeHandle = widgetOpened ? (
-            <div className='sk-close-handle sk-close-hidden'>
-                <i className='fa fa-times'></i>
-            </div>
-            ) : (
-            <div className='sk-show-handle sk-appear-hidden'>
-                <i className='fa fa-arrow-up'></i>
-            </div>
-            );
+        const closeHandle = embedded ?
```
Comment (mspensieri) [inline src/js/components/header.jsx]: not too crazy about this double ternary

File: src/js/smooch.jsx [hunk 2]
```diff
@@ -212,24 +219,41 @@ export class Smooch {
...
+        let {embedded} = store.getState().appState;
         disconnectFaye();
         store.dispatch(reset());
+        if (process.env.NODE_ENV !== 'test') {
+            unmountComponentAtNode(this._container);
+        }
+
+        if (embedded) {
+            // retain the embed mode
+            store.dispatch(AppStateActions.setEmbedded(true));
```
Comment (mspensieri) [inline src/js/smooch.jsx]: What's happening here? Destroy won't remove the element for embedded mode?

**Comment by mspensieri** [comment]
:+1:

File: src/js/components/header.jsx (hunk 1 above)
Comment (lemieux) [inline src/js/components/header.jsx]: yeah, I'll clean that up

File: src/js/smooch.jsx (hunk 2 above)
Comment (lemieux) [inline src/js/smooch.jsx]: In the non-embedded mode, the container is created by us and needs to be removed from the DOM. In the embedded version, the container is provided and we don't have control over it.

The real "remove from DOM" action happens couple of line above with `unmountComponentAtNode(this._container);` which remove the actual widget from the DOM and initiate the unmount flow for all components in it.

Comment (mspensieri) [inline src/js/smooch.jsx]: Got it, thanks

**Comment by spasiu** [comment]
:+1:

File: src/js/smooch.jsx [hunk 3]
```diff
@@ -212,24 +219,41 @@ export class Smooch {
...
-        document.body.removeChild(this._el);
         delete this.appToken;
-        delete this._el;
+        delete this._container;
         observable.trigger('destroy');
     }
 
     open() {
-        store.dispatch(openWidget());
+        let {embedded} = store.getState().appState;
```
Comment (alavers) [inline src/js/smooch.jsx]: perhaps we should introduce a `isEmbedded()` helper function?

**Pull request merged**